src/session_impl.cpp

```diff
@@ -1938,7 +1949,7 @@ namespace aux {
 
 		if (map_ports)
 		{
-			for (auto const& s : m_listen_sockets)
+			for (auto const& s : new_sockets)
```

Comment:
Based on the documentation of `session_handle::reopen_network_sockets` I take it that the `map_ports` is intended to re-map all listen sockets, including existing ones. In that case this should loop over the existing sockets rather than the new ones. That would allow the `if (!map_ports)` condition to be removed above when mapping uPNP for new sockets, and fix a bug I just noticed which causes NATPMP to be mapped twice for new sockets.